From: Vicentiu Ciorbaru <cvicentiu@gmail.com>

Subject: Workaround for ICE in mroonga on Arm64 architecture
Details: There seems to be a bug in gcc-6.3.0 on arm64 where it fails
to compile a file which makes use of the not_equal_float function.
A basic logic workaround solves the problem by calling the equal_float
function instead.


diff --git a/storage/mroonga/vendor/groonga/lib/ts/ts_expr_node.c b/storage/mroonga/vendor/groonga/lib/ts/ts_expr_node.c
index 44378cfa..a3d838c3 100644
--- a/storage/mroonga/vendor/groonga/lib/ts/ts_expr_node.c
+++ b/storage/mroonga/vendor/groonga/lib/ts/ts_expr_node.c
@@ -562,7 +562,7 @@ inline static grn_ts_bool
 grn_ts_op_not_equal_float(grn_ts_float lhs, grn_ts_float rhs)
 {
   /* To suppress warnings, "lhs != rhs" is not used. */
-  return (lhs < rhs) || (lhs > rhs);
+  return !grn_ts_op_equal_float(lhs, rhs);
 }
 
 /* grn_ts_op_not_equal_time() returns lhs != rhs. */
